extension/src/background/messageListener/freighterApiMessageListener.ts

```diff
@@ -183,33 +182,23 @@ export const freighterApiMessageListener = (
 
       const isValidatingMemo =
         (await getIsMemoValidationEnabled()) && isMainnet;
-      const isValidatingSafety =
-        (await getIsSafetyValidationEnabled()) && isMainnet;
 
-      if (isValidatingMemo || isValidatingSafety) {
+      if (isValidatingMemo) {
         _operations.forEach((operation: StellarSdk.Operation) => {
           accountData.forEach(
             ({ address, tags }: { address: string; tags: string[] }) => {
               if (
                 "destination" in operation &&
                 address === operation.destination
               ) {
-                let collectedTags = [...tags];
+                const collectedTags = [...tags];
```

Comment:
If this no longer gets mutated, then is it just `tags` now?

Reply:
It can still be mutated because we're filtering if `!isValidatingMemo` on ln 197. It's good practice to not mutate a passed param

Reply:
hmm that seems like a bug on 197 then, filter does not mutate the array you are filtering, it makes a shallow copy so on 197 we're not assigning the result of that filter to anything and `collectedTags` is not mutated there.

Reply:
Oh man - you're right. Let me take a closer look 👀

Reply:
Addressed in the latest commit. Looks like we had been reassigning the var correctly for the other safety checks, but not for memo required 🤦

Reply:
Thanks, lgtm!